ngraph/core/reference/include/ngraph/runtime/reference/scatter_update.hpp

```diff
@@ -1,86 +1,122 @@
-// Copyright (C) 2020 Intel Corporation
-// SPDX-License-Identifier: Apache-2.0
+//*****************************************************************************
+// Copyright 2017-2020 Intel Corporation
 //
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
 
 #pragma once
 
-#include <string>
+#include <cstring>
+
+#include "ngraph/check.hpp"
 #include "ngraph/coordinate_transform.hpp"
 #include "ngraph/shape.hpp"
 
-using namespace ngraph;
-
 namespace ngraph
 {
     namespace runtime
     {
         namespace reference
         {
-            template <typename dataType, typename indicesType, typename axisType>
-            void scatterUpdate(const dataType* inputData,
-                               const indicesType* indices,
-                               const dataType* updates,
-                               const axisType* _axis,
-                               dataType* outBuf,
-                               const Shape& dataShape,
-                               const Shape& indicesShape,
-                               const Shape& updatesShape)
+            void scatter_update(const char* input_data,
+                                const int64_t* indices,
+                                const char* updates,
+                                const int64_t& axis,
```

Comment:
It's not very important and you can adress this only if you need to do more changes in this file but in general PODs should be passed by value rather than by reference.

Reply:
I agree, done